input-explainer.md

```diff
@@ -280,6 +292,91 @@ When the canvas receives a `pointerdown` event an `XRInputSource` is created wit
+  - The `Gamepad`'s `index` attribute must be `-1`.
+  - The `Gamepad`'s `connected` attribute must always be `true`.
+  - The `Gamepad`'s `mapping` attribute must be `xr-standard`.
+
+Finally, the `id` attribute for `Gamepad`s surfaced by the WebXR API are more strictly formatted than those of traditional gamepads in order to make them a more appropriate key for determining rendering assets.
+
+  - The `id` MAY be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason. Applications should render a generic input device in this case.
+  - Inline sessions MUST only expose `id`s of `'unknown'`.
+  - Otherwise the `id` should be a lower-case string that describes the physical input source.
+    - For most devices this SHOULD be of the format `<vendor>-<product-id>`. For example: `oculus-touch`. UAs SHOULD make an effort to align on the strings that are returned for any given device.
+    - It should not include an indication of the handedness of the input source, as that should be implied by the `handedness` attribute.
+
+All other attributes behave as described in the [Gamepad](https://w3c.github.io/gamepad/) specification.
+
+The WebXR Device API introduces a new standard controller layout indicated by the `mapping` value of `xr-standard`. (Additional mapping variants may be added in the future if necessary.) This defines a specific layout for the inputs most commonly found on XR controller devices today. The following table describes the buttons/axes and their physical locations:
+
+| Button/Axis  | Location                          |
+| ------------ | --------------------------------- |
+| buttons[0]   | Primary trigger                   |
+| buttons[1]   | Primary Touchpad/Joystick click   |
+| buttons[2]   | Grip/Secondary trigger            |
+| buttons[3]   | Secondary Touchpad/Joystick click |
+| axes[0]      | Primary Touchpad/Joystick X       |
+| axes[1]      | Primary Touchpad/Joystick Y       |
+| axes[2]      | Secondary Touchpad/Joystick X     |
+| axes[3]      | Secondary Touchpad/Joystick Y     |
+
+Additional device-specific inputs may be exposed after these reserved indices, but devices that lack one of the canonical inputs must still preserve their place in the array. If a device has both a touchpad and a joystick the UA should designate one of them to be the primary axis-based input and expose the other at axes[2] and axes[3] with an associated button at button[3].
+
+```js
+function onXRFrame(timestamp, frame) {
+  let inputSource = primaryInputSource;
+
+  // Check to see if the input source has buttons/axes and is using the standard
+  // mapping.
+  if (inputSource && inputSource.gamepad &&
+      inputSource.gamepad.mapping == 'xr-standard') {
+    let gamepad = inputSource.gamepad;
+    
+    // Use joystick or touchpad values for movement.
+    MoveUser(gamepad.axes[0], gamepad.axes[1]);
```

Comment:
I swear I saw a version in which you'd fixed this sample so it didn't assume there were axes when that may not necessarily be true...?